### PR TITLE
Add service provider to cloud realm

### PR DESCRIPTION
**Pull request by eiffel777**
This PR adds the provider group by to the cloud realm. This group by is needed for the scenario where a cloud resource is comprised of more than one installation of a cloud computing system. The service provider is specified by using a config variable named ${SERVICE_PROVIDER} when using the etl_overseer.php script to run an etl process.

Like the provider group by in the jobs realm, the group by for provider in the cloud realm will be disabled by default as this is something that is primarily needed for XSEDE and can be enabled in the XSEDE module.

## Tests performed
Manually tested and ran all unit, regression, component and integration tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project as found in the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: classes/DataWarehouse/Query/Cloud/GroupBys/GroupByServiceProvider.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+class GroupByServiceProvider extends \DataWarehouse\Query\Cloud\GroupBy
+{
+    public static function getLabel()
+    {
+        return 'Provider';
+    }
+
+    public function getInfo()
+    {
+        return 'The location of the resource.';
```
Comment (chakrabortyr) [inline classes/DataWarehouse/Query/Cloud/GroupBys/GroupByServiceProvider.php]: Can we get a more thorough description than this? It doesn't tell me much about what it is.

File: configuration/etl/etl_tables.d/cloud_common/cloudfact_by_.json [hunk 2]
```diff
@@ -110,12 +110,16 @@
...
+            },{
                 "name": "domain_id",
                 "type": "int(11)",
                 "nullable": true,
                 "comment": "DIMENSION: A domain is a high-level container for projects, users and groups."
+            },{
+                "name": "service_provider",
+                "type": "int(11)",
+                "nullable": true,
+                "comment": "DIMENSION: The orgainization that is hosting a virtual machine."
```
Comment (chakrabortyr) [inline configuration/etl/etl_tables.d/cloud_common/cloudfact_by_.json]: This isn't consistent with how it's being defined in the group by.

**Comment by eiffel777** [comment]
@chakrabortyr Description updates made. It should match how service provider is described in the Jobs realm group by.

As for the sanity check, it turns I didn't need to change the reconstructor since the instance id used is the id that is auto generated on the instance table not the openstack instance uuid. The instance id from the instance table already handles the case same openstack instance uuid but different service providers.

**Comment by chakrabortyr** [comment]
@eiffel777 Okay, LGTM once the tests pass.

**Comment by eiffel777** [comment]
@plessbd Tests have been added.

**Comment by eiffel777** [comment]
@plessbd The tests pass now

File: configuration/etl/etl_tables.d/cloud_common/cloudfact_by_.json [hunk 3]
```diff
@@ -116,6 +116,11 @@
                 "type": "int(11)",
                 "nullable": true,
                 "comment": "DIMENSION: A domain is a high-level container for projects, users and groups."
+            },{
```
Comment (ryanrath) [inline configuration/etl/etl_tables.d/cloud_common/cloudfact_by_.json]: So just a quick question, I notice that in:
- `cloud_common/cloudfact_by_.json`
- `cloud_common/event.json`
- `cloud_common/instance.json`
- `cloud_common/session_records.json`
- `cloud_generic/raw_event.json`
- `cloud_generic/staging_event.json`
- `cloud_openstack/raw_event.json`
- `cloud_openstack/staging_event.json `

Where the type is `int(11)` the `nullable` and `default` values vary from file to file. Is this something that's required due to ETL / data constrictions? Or could having them be consistent possibly be a thing?

**Pull request merged**